### PR TITLE
Update leadership lock handling

### PR DESCRIPTION
**Pull request by cjus**
- Extend leadership lock interval
- If same instance of moose was the leader and lost leaderhip then if it renewal leadership it should not register cron jobs.

### CODE REVIEW

File: apps/framework-cli/src/cli/routines.rs [hunk 1]
```diff
@@ -362,24 +366,32 @@ async fn manage_leadership_lock(
...
-        let mut client = redis_client.lock().await;
-        if let Err(e) = client.broadcast_message("leader.new").await {
-            error!(
-                "<RedisClient> Failed to broadcast new leader message: {}",
-                e
-            );
         }
+    } else {
+        // If we lost the lock, reset the leadership tasks flag
+        IS_RUNNING_LEADERSHIP_TASKS.store(false, Ordering::SeqCst);
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/routines.rs:400]: i think `false` should be set after the spawned tasks inside `leadership_tasks` are dead

File: apps/framework-cli/src/cli/routines.rs [hunk 2]
```diff
@@ -362,24 +366,32 @@ async fn manage_leadership_lock(
...
-
-        let project_clone = project.clone();
-        tokio::spawn(async move {
-            if let Err(e) = leadership_tasks(project_clone).await {
-                error!("<RedisClient> Error executing leadership tasks: {}", e);
+    if has_lock {
+        if is_new_acquisition && !IS_RUNNING_LEADERSHIP_TASKS.load(Ordering::SeqCst) {
+            info!("<RedisClient> Obtained leadership lock, performing leadership tasks");
+
+            IS_RUNNING_LEADERSHIP_TASKS.store(true, Ordering::SeqCst);
```
Comment (callicles) [inline apps/framework-cli/src/cli/routines.rs]: Do you set this back to false when the leadership task is lost?

File: apps/framework-cli/src/cli/routines.rs (hunk 1 above)
Comment (cjus) [reply apps/framework-cli/src/cli/routines.rs:400]: Yes nice catch updating PR

File: apps/framework-cli/src/cli/routines.rs (hunk 2 above)
Comment (cjus) [reply apps/framework-cli/src/cli/routines.rs]: Yes updating PR

**Review by phiSgr — APPROVED** [review]
lgtm

File: apps/framework-cli/src/cli/routines.rs [hunk 3]
```diff
@@ -384,6 +388,13 @@ async fn manage_leadership_lock(
                 e
             );
         }
+    } else if IS_RUNNING_LEADERSHIP_TASKS.load(Ordering::SeqCst) {
+        // Stop the CronRegistry first
+        if let Err(e) = CronRegistry::new().await?.stop().await {
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/routines.rs]: use the `cron_registry` passed in

**Review by phiSgr — DISMISSED** [review]

File: apps/framework-cli/src/cli/routines.rs (hunk 3 above)
Comment (cjus) [reply apps/framework-cli/src/cli/routines.rs]: updated

File: apps/framework-cli/src/cli/routines.rs [hunk 4]
```diff
@@ -384,6 +388,12 @@ async fn manage_leadership_lock(
                 e
             );
         }
+    } else if IS_RUNNING_LEADERSHIP_TASKS.load(Ordering::SeqCst) {
```
Comment (HDIBen) [inline apps/framework-cli/src/cli/routines.rs:395]: Where / how does IS_RUNNING_LEADERSHIP_TASKS becomes true? It's initialized as false on line 132 and is set to false on line 396 but I don't see where it's going to be true. Thanks for explaining.

Comment (cjus) [reply apps/framework-cli/src/cli/routines.rs:395]: This is correct and is an issue.  Updating PR now.

File: apps/framework-cli/src/cli/routines.rs [hunk 5]
```diff
@@ -373,11 +373,14 @@ async fn manage_leadership_lock(
...
         info!("<RedisClient> Obtained leadership lock, performing leadership tasks");
 
+        IS_RUNNING_LEADERSHIP_TASKS.store(true, Ordering::SeqCst);
+
         let project_clone = project.clone();
         let cron_registry: CronRegistry = cron_registry.clone();
         tokio::spawn(async move {
             if let Err(e) = leadership_tasks(project_clone, cron_registry).await {
                 error!("<RedisClient> Error executing leadership tasks: {}", e);
+                IS_RUNNING_LEADERSHIP_TASKS.store(false, Ordering::SeqCst);
```
Comment (callicles) [inline apps/framework-cli/src/cli/routines.rs]: Feels like this should be outside of the `if`

**Review by callicles — COMMENTED** [review]
Maybe last comment?

File: apps/framework-cli/src/cli/routines.rs (hunk 5 above)
Comment (cjus) [reply apps/framework-cli/src/cli/routines.rs]: yes fixed

**Review by callicles — APPROVED** [review]

**Pull request merged**